From ef5f6ed04f31eaa14fa0c5712694bd4b34b76e53 Mon Sep 17 00:00:00 2001
From: Stef Walter <stefw@gnome.org>
Date: Fri, 15 Feb 2013 14:36:59 +0100
Subject: [PATCH] dbus: Don't reverse the order of items returned from
 SearchItems()

Applications expect the latest item to be returned first. We sort
the items carefully. Fix regression which reverses them.

https://bugzilla.gnome.org/show_bug.cgi?id=693884
---
 daemon/dbus/gkd-secret-objects.c | 3 +++
 1 file changed, 3 insertions(+)

diff --git a/daemon/dbus/gkd-secret-objects.c b/daemon/dbus/gkd-secret-objects.c
index 8bc500f..b42c7be 100644
--- a/daemon/dbus/gkd-secret-objects.c
+++ b/daemon/dbus/gkd-secret-objects.c
@@ -424,6 +424,9 @@ item_cleanup_search_results (GckSession *session, GList *items,
 
 		g_free (value);
 	}
+
+	*locked = g_list_reverse (*locked);
+	*unlocked = g_list_reverse (*unlocked);
 }
 
 static DBusMessage*
-- 
1.8.1